test/FileWriterTest.php

```diff
@@ -140,11 +147,33 @@ public function testMultipleWriters(string $writer, bool $expectedResult) : void
         }
     }
 
+    public function testUnwritableDirThrowsException()
+    {
+        $this->expectException(RuntimeException::class);
+        $dir = sys_get_temp_dir() . '/unwritable';
+        mkdir($dir);
+        chmod($dir, 0400);
+        FileWriter::writeFile($dir . '/test', 'foo');
```

Comment:
I see we are getting issue on windows as we can't remove that directory.
I wonder if try ... finally and change permission will work here?

```php
try {
    FileWriter::writeFile($dir . '/test', 'foo');
} finally {
    chmod($dir, 0700);
}
```